### ISSUE TITLE
Rainbow does not start with the initial configuation

### ISSUE THREAD

**Issue by schmerl** [cmu-mars/issues#81]
I don't think rainbow gets the configuration of the first task. This is probably because the system announces the configuration before rainbow starts. This needs to be coordinated somehow.

This would lead to an error where the battery analyzer is not going to say there is no power if the battery is too low in the first task.

**Comment by pooyanjamshidi** [comment]
@schmerl by the configuration of the task, you mean configuration of the robot in the first task? I have a configuration option for that our config manager backend.

**Comment by schmerl** [comment]
The problem is that if you are announcing the config to a topic, that
announcement happens before Rainbow has started and so it is not known
until it is changed again. I think this is what is happening, but need to
investigate it further.

**Comment by pooyanjamshidi** [comment]
No, it's not topic based, it simply a service that you can call anytime and ask what is the configuration of the robot at this very exact moment. I think we discussed this before, but anyway, let me know if I need to change any part of the code, it is all documented int he config manager repo.

**Comment by schmerl** [comment]
OK. I'll check to see why the battery analyzer is running only after the
second instruction.

**Comment by schmerl** [comment]
This is a test of notifications.

**Comment by pooyanjamshidi** [comment]
@schmerl is this now resolved? what's the status?

**Comment by ivoysey** [comment]
@schmerl @pooyanjamshidi what's the status of this issue? did it get resolved last week? does it belong in RR3?

**Comment by schmerl** [comment]
@pooyanjamshidi  I do not think it is resolved. You should be able to come up with a test case that would see if it is a problem. I think the problem could be demonstrated by setting an initial battery perturbation that doesn't give enough battery to complete the first task. I will wait for your confirmation of whether this is a problem.

**Comment by pooyanjamshidi** [comment]
If this is a rainbow problem, then this needs to be resolved whether it manifests an error in one scenario or not. The default configuration where the robot is running is communicated with Ranbow/planner via the json file which is shared. Also, at anytime, rainbow can query what configuration the robot is in. So I this must be fixed in Rainbow if it is already not?

**Comment by pooyanjamshidi** [comment]
@javier-camara can you please confirm whether the planner assuming/using the default configuration correctly?

**Comment by schmerl** [comment]
@pooyanjamshidi <https://github.com/pooyanjamshidi> Javier cannot confirm
this because it is not his problem. You should come up with a test that
will illustrate if this is an error or not, and run it. Note that neither I
nor Javier are running tests of CP1 currently - we are expecting you to run
these tests for the whole system and report any bugs/observations to us. I
reported this bug because this was something that I observed pre-RR2 but
could not confirm properly.

Please note that you are responsible for CP1 testing -- do not assume that
we are doing any testing other than what we have already done. As the
observation clearly states, this might be a problem with the integration
between Rainbow and your system, or it may not be a problem at all. Please
confirm if you can replicate it in your own integration testing and then we
can follow up.

**Comment by pooyanjamshidi** [comment]
@schmerl how can I confirm something that I don't have access?? I should have an API to confirm whether Rainbow pick up the default configuration, does rainbow expose what configuration it is assuming, how can I access it? besides we are working as a team, there is no boundary of challenge problem whatever, if you see a bug and since this is obviously related to the Rainbow assuming a configuration, then you should have already fixed it, right? or I'm missing something? I would have done the same for you if you were using one of the components that I have developed. In the description you are mentioning about coordination, then you said about the topic, I clarified that, so rainbow can read the default configuration either through the json file or querying the cp1 backend, how currently it does that?

**Comment by schmerl** [comment]
@pooyanjamshidi I gave you an example of how you could test that above. So you should run that test. Yes, we are working as a team but you agreed to manage (i.e., drive) CP1, and so I am expecting you to manage (i.e., drive) CP1. I am not asking you to fix the problem, I am asking you to check if you see the problem, produce some logs, so that we can do debugging. It is not "obviously" related to Rainbow. It may be related to Rainbow, it may be related to the interface between Rainbow and your system - it's probably not related to planning so it is probably not something that he can confirm. So, I'm asking you _do you see the problem in your integration testing of CP1?_

**Comment by pooyanjamshidi** [comment]
@schmerl I think it picks up the first config correctly as far as I can see in the command line, can you check out the logs? as you suggested I set a very low battery for the first task that it cannot finish it....

[rainbow-analyzers.log](https://github.com/cmu-mars/issues/files/2142378/rainbow-analyzers.log)
[rainbow-logging.log](https://github.com/cmu-mars/issues/files/2142379/rainbow-logging.log)
